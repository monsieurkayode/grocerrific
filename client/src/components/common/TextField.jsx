import React from 'react';
import {
  oneOfType,
  string,
  number,
  func
} from 'prop-types';

const TextField = ({
  name,
  label,
  placeholder,
  type,
  error,
  value,
  field,
  onChange
}) => (
  <div className={field}>
    <label htmlFor={name}>{label}&nbsp;<span>*</span></label>
    <input
      type={type}
      name={name}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      className={error.length > 0 && 'alert'}
    />
    { error.length > 0 && <div className="error">{error}</div> }
  </div>
);

TextField.defaultProps = {
  type: 'text',
  field: 'form-field',
  value: '',
  error: ''
};

TextField.propTypes = {
  name: string.isRequired,
  label: string.isRequired,
  placeholder: string.isRequired,
  type: string,
  error: string,
  value: oneOfType([string, number]),
  field: string,
  onChange: func.isRequired
};

export default TextField;
