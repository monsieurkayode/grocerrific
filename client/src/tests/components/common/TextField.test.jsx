import React from 'react';
import { mount } from 'enzyme';
import TextField from '../../../components/common/TextField';

const setup = () => {
  const props = {
    name: '',
    label: '',
    placeholder: '',
    field: '',
    onChange: jest.fn(),
    onFocus: jest.fn()
  };

  const wrapper = mount(<TextField {...props} />);

  return { wrapper, props };
};

describe('TextField Component', () => {
  const { wrapper, props } = setup();

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChange and onFocus function on text input', () => {
    const textInput = wrapper.find('input');
    textInput.simulate('change', { value: 'Hello' });
    expect(props.onChange).toHaveBeenCalled();
  });

  it('should have an error field', () => {
    const errorDiv = wrapper.find('.error');
    expect(errorDiv.length).toBe(1);
  });

  it('should set the visiblity of error div to visible if errors', () => {
    wrapper.setProps({ error: 'Error' });
    const errorDiv = wrapper.find('.error');
    expect(errorDiv.text()).toContain('Error');
    expect(errorDiv.props().style.visibility).toBe('visible');
  });
});
