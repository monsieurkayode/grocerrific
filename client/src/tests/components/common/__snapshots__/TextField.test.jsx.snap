// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Component renders correctly 1`] = `
<TextField
  error=""
  field=""
  label=""
  name=""
  onChange={[MockFunction]}
  onFocus={[MockFunction]}
  placeholder=""
  type="text"
  value=""
>
  <div
    className=""
  >
    <label
      htmlFor=""
    >
       
      <span>
        *
      </span>
    </label>
    <input
      className=""
      name=""
      onChange={[MockFunction]}
      onFocus={[MockFunction]}
      placeholder=""
      type="text"
      value=""
    />
    <div
      className="error"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    >
       
    </div>
  </div>
</TextField>
`;
